pkg/virt-controller/watch/pool/pool_test.go

```diff
@@ -192,6 +193,53 @@ var _ = Describe("Pool", func() {
 			}
 		}
 
+		createVMsWithOrdinal := func(pool *poolv1.VirtualMachinePool, count int, newPoolRevision *appsv1.ControllerRevision, oldPoolRevision *appsv1.ControllerRevision, vm *v1.VirtualMachine) {
+			for i := 0; i < 5; i++ {
+				vmCopy := vm.DeepCopy()
+				vmCopy.Name = fmt.Sprintf("%s-%d", pool.Name, i)
+				vmCopy.CreationTimestamp = metav1.Time{Time: metav1.Now().Add(-time.Duration(5-i) * time.Second)}
+				vmCopy = injectPoolRevisionLabelsIntoVM(vmCopy, newPoolRevision.Name)
+				markVmAsReady(vmCopy)
+				vmi := createReadyVMI(vmCopy, oldPoolRevision)
+				addVM(vmCopy)
+				addVMI(vmi)
+			}
+		}
+
+		addNodeSelectorToVM := func(vms []*v1.VirtualMachine, pool *poolv1.VirtualMachinePool) {
+			for _, vm := range vms {
+				if vm.Name == fmt.Sprintf("%s-%d", pool.Name, 0) || vm.Name == fmt.Sprintf("%s-%d", pool.Name, 1) {
+					if vm.Spec.Template.Spec.NodeSelector == nil {
+						vm.Spec.Template.Spec.NodeSelector = make(map[string]string)
+					}
+					vm.Spec.Template.Spec.NodeSelector["node-role.kubernetes.io/worker"] = "test"
+					controller.vmIndexer.Update(vm)
+				}
+			}
+		}
+
+		addLabelToVM := func(vms []*v1.VirtualMachine, pool *poolv1.VirtualMachinePool) {
+			for _, vm := range vms {
+				if vm.Name == fmt.Sprintf("%s-%d", pool.Name, 0) || vm.Name == fmt.Sprintf("%s-%d", pool.Name, 1) {
+					if vm.Spec.Template.ObjectMeta.Labels == nil {
+						vm.Spec.Template.ObjectMeta.Labels = make(map[string]string)
+					}
+					vm.Spec.Template.ObjectMeta.Labels["app"] = "test"
+					controller.vmIndexer.Update(vm)
+				}
+			}
+		}
+
+		expectPoolUpdate := func(replicas int32) {
+			fakeVirtClient.Fake.PrependReactor("update", "virtualmachinepools", func(action k8stesting.Action) (handled bool, obj runtime.Object, err error) {
+				update, ok := action.(k8stesting.UpdateAction)
+				Expect(ok).To(BeTrue())
+				updateObj := update.GetObject().(*poolv1.VirtualMachinePool)
+				Expect(updateObj.Status.Replicas).To(Equal(replicas))
+				return true, update.GetObject(), nil
+			})
+		}
```

Comment:
Same, instead of overwriting the reactor you could get the pool after the call to `sanityExecutte` and assert that the replica count matches.